Source/TestDafny/MultiBackendTest.cs

```diff
@@ -339,15 +357,70 @@ int ExpectedExitCode
       dafnyArgs = dafnyArgs.Concat(new[] { "--include-runtime:false", "--input", runtimePath });
     }
 
-    var (exitCode, outputString, error) = await RunDafny(options.DafnyCliPath, dafnyArgs);
+    int exitCode;
+    string outputString;
+    string error;
+    try {
+      (exitCode, outputString, error) = await RunDafny(options.DafnyCliPath, dafnyArgs);
+    } catch (Exception e) {
+      (exitCode, outputString, error) = (3, e.ToString(), e.ToString());
+    }
+
     var compilationOutputPrior = new Regex("\r?\nDafny program verifier[^\r\n]*\r?\n").Match(outputString);
     if (compilationOutputPrior.Success) {
       outputString = outputString.Remove(0, compilationOutputPrior.Index + compilationOutputPrior.Length);
     }
 
+    var diffMessage = exitCode == 0 ? AssertWithDiff.GetDiffMessage(expectedOutput, outputString) : null;
+    if (checkFile == null && exitCode != 0) {
+      if (Environment.GetEnvironmentVariable("DAFNY_INTEGRATION_TESTS_UPDATE_TARGET_EXPECT_FILE") == "true") {
+        if ((Environment.GetEnvironmentVariable("DAFNY_INTEGRATION_TESTS_ROOT_DIR") ?? "") == "") {
+          await output.WriteLineAsync(
+            "DAFNY_INTEGRATION_TESTS_UPDATE_TARGET_EXPECT_FILE is true but DAFNY_INTEGRATION_TESTS_ROOT_DIR is not set");
+        } else {
+          // Ensure we don't accidentally do this for every compiler.
+          var sourcePath = Path.Join(Environment.GetEnvironmentVariable("DAFNY_INTEGRATION_TESTS_ROOT_DIR"),
+            CheckFileForBackend(options, backend));
+          var contentCheck = (outputString + "\n" + error).Trim();
+          var checkOutput = "";
+          if (contentCheck == "") {
+            checkOutput = string.Join("\n", expectedOutput.Split("\n").Select(line => "// CHECK-NOT: " + line));
+          } else {
+            checkOutput = string.Join("\n",
+              contentCheck.Split("\n").Select(line => "// CHECK-L: " + line));
+          }
+
+          if (!File.Exists(sourcePath)) {
+            await File.WriteAllTextAsync(sourcePath, checkOutput);
+          } else {
+            await output.WriteLineAsync($"Check file existed but was not imported yet. Ignoring.");
+          }
+
+          await output.WriteLineAsync($"Please modify the new check file {sourcePath} so that it's valid no matter what.");
+          await output.WriteLineAsync(outputString);
+          await output.WriteLineAsync("Error:");
+          await output.WriteLineAsync(error);
+
+          return exitCode;
+        }
+      }
+    }
+    
     if (exitCode == 0) {
-      var diffMessage = AssertWithDiff.GetDiffMessage(expectedOutput, outputString);
       if (diffMessage == null) {
+        if (checkFile != null) {
+          // The test now works, we delete the check file
+          if (Environment.GetEnvironmentVariable("DAFNY_INTEGRATION_TESTS_UPDATE_TARGET_EXPECT_FILE") == "true") {
```

Comment:
Could we name this just `DAFNY_INTEGRATION_TESTS_UPDATE_EXPECT_FILES` and make it switch on `DiffCommand.UpdateExpectFile` too?